must-gather/collection-scripts/ceph_commands.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+base_collection_path=$1
+namespace=$2
+if [ "${base_collection_path}" = "" ];then
+    echo "Base collection path for ceph commands is not passed. Exiting."
+    exit 0
+fi
+
+if [ "${namespace}" = "" ];then
+    echo "Namespace for ceph commands is not passed. Exiting."
+    exit 0
+fi
+
+# Ceph commands
+ceph_commands=()
+ceph_commands+=("ceph status")
+ceph_commands+=("ceph health detail")
+ceph_commands+=("ceph osd tree")
+ceph_commands+=("ceph osd stat")
+ceph_commands+=("ceph osd dump")
+ceph_commands+=("ceph mon stat")
+ceph_commands+=("ceph mon dump")
+ceph_commands+=("ceph df")
+ceph_commands+=("ceph report")
+ceph_commands+=("ceph osd df tree")
+ceph_commands+=("ceph fs ls")
+ceph_commands+=("ceph pg dump")
+ceph_commands+=("ceph osd crush show-tunables")
+ceph_commands+=("ceph osd crush dump")
+ceph_commands+=("ceph mgr dump")
+ceph_commands+=("ceph mds stat")
+ceph_commands+=("ceph versions")
+ceph_commands+=("ceph fs dump")
+ceph_commands+=("ceph auth list")
+
+# Ceph volume commands
+ceph_volume_commands+=()
+ceph_volume_commands+=("ceph-volume lvm list")
+
+ceph_commands_output_dir="${base_collection_path}/must_gather_commands"
+mkdir -p "${ceph_commands_output_dir}"
+ceph_commands_json_output_dir="${ceph_commands_output_dir}/json_output"
+mkdir -p "${ceph_commands_json_output_dir}"
+
+# Collecting output of ceph commands
+cpu_numbers=$(nproc 2>/dev/null || echo 4)
+helper_pod_status=$(timeout 120 oc get pods "${HOSTNAME}"-helper -n "${namespace}" -o jsonpath="{.status.phase}")
+if [ "${helper_pod_status}" = "Running" ]; then
+    for ((i = 0; i < ${#ceph_commands[@]}; i++)); do
+        while true; do
+            if [ "$(jobs -rp | wc -l)" -lt "${cpu_numbers}" ]; then break; fi
+            sleep 3
+        done
+        printf " -> Fetching output for '%s' from %s pod\n"  "${ceph_commands[$i]}" "${HOSTNAME}-helper"
+        ceph_command_output_file=${ceph_commands_output_dir}/${ceph_commands[$i]// /_}
+        ceph_command_json_output_file=${ceph_commands_json_output_dir}/${ceph_commands[$i]// /_}_--format_json-pretty
+        { timeout 120 oc -n "${namespace}" exec  "${HOSTNAME}"-helper -- bash -c "${ceph_commands[$i]} --connect-timeout=15"  >> "${ceph_command_output_file}"; } 2>/dev/null &
```

Comment:
Piping stderr to dev/null could also result in silent failures. Could you reasonably pipe it to its own file for debugging purposes? In most cases it won't matter, but in rare ones it might save someone a lot of trouble.